util/defaulting/defaulting.go

```diff
@@ -1,68 +0,0 @@
-/*
```

Comment:
Wow, this was entirely unused?

There's an entry in golanci lint about it which maybe also could go away?

Reply:
In core CAPI we migrated to custom validator/defaulter a long time ago, and they have their own equivalent of these utils.

These might still have been used by providers though. But as Defaulter/Validator have been dropped in CR there is no way for us to keep these utils.

Reply:
Nice catch regareding golangci lint, dropped